app/models/workflow_step.rb

```diff
@@ -39,11 +40,21 @@ def set_completed_at
     self.completed_at = Time.now
   end
 
+  # Check if step's status is a completed one *and* that there is no completed_at value
+  def completed_and_not_already_completed?
+    completed? && !already_completed?
+  end
+
   ##
   # indicate if this step is marked as completed
   # @return [boolean]
   def completed?
-    COMPLETED_STATES.include? status
+    COMPLETED_STATES.include?(status)
```

Comment:
any reason parens were added?  is there a preferred rubocop style?

Reply:
Ah, because I was originally going to add a condition to this method and didn't want the parser to get confused, but then I backed it out and left the parens.